homeassistant/components/airzone/select.py

```diff
@@ -0,0 +1,160 @@
+    API_HEAT_ANGLE,
+    API_SLEEP,
+    AZD_COLD_ANGLE,
+    AZD_HEAT_ANGLE,
+    AZD_NAME,
+    AZD_SLEEP,
+    AZD_ZONES,
+)
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+
+@dataclass
+class AirzoneSelectDescriptionMixin:
+    """Define an entity description mixin for select entities."""
+
+    api_param: str
+    options_dict: dict[str, int]
+
+
+@dataclass
+class AirzoneSelectDescription(SelectEntityDescription, AirzoneSelectDescriptionMixin):
+    """Class to describe an Airzone select entity."""
+
+
+GRILLE_ANGLE_DICT: Final[dict[str, int]] = {
+    "90º": GrilleAngle.DEG_90,
+    "50º": GrilleAngle.DEG_50,
+    "45º": GrilleAngle.DEG_45,
+    "40º": GrilleAngle.DEG_40,
+}
+
+SLEEP_DICT: Final[dict[str, int]] = {
+    "Off": SleepTimeout.SLEEP_OFF,
+    "30m": SleepTimeout.SLEEP_30,
+    "60m": SleepTimeout.SLEEP_60,
+    "90m": SleepTimeout.SLEEP_90,
+}
+
+
+ZONE_SELECT_TYPES: Final[tuple[AirzoneSelectDescription, ...]] = (
+    AirzoneSelectDescription(
+        api_param=API_COLD_ANGLE,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_COLD_ANGLE,
+        name="Cold Angle",
+        options_dict=GRILLE_ANGLE_DICT,
+    ),
+    AirzoneSelectDescription(
+        api_param=API_HEAT_ANGLE,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_HEAT_ANGLE,
+        name="Heat Angle",
+        options_dict=GRILLE_ANGLE_DICT,
+    ),
+    AirzoneSelectDescription(
+        api_param=API_SLEEP,
+        entity_category=EntityCategory.CONFIG,
+        key=AZD_SLEEP,
+        name="Sleep",
+        options_dict=SLEEP_DICT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[AirzoneSelect] = []
+
+    for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in ZONE_SELECT_TYPES:
+            if description.key in zone_data:
+                _desc = replace(
+                    description,
+                    options=list(description.options_dict.keys()),
+                )
+                entities.append(
+                    AirzoneZoneSelect(
+                        coordinator,
+                        _desc,
+                        entry,
+                        system_zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class AirzoneSelect(AirzoneEntity, SelectEntity):
```

Comment:
```suggestion
class AirzoneBaseSelect(AirzoneEntity, SelectEntity):
```

It took me a minute to figure out this was a base class for your Zone entities.

Reply:
Done in ed144810a6f7bfef84998ed3747d44fc49e6ff17